### PR TITLE
User auth register and login

### PR DESCRIPTION
**Pull request by yliu0715**
Adds register and login pages and a navbar
Fixes https://github.com/meowl-surveillance-system/meowl/issues/47 and https://github.com/meowl-surveillance-system/meowl/issues/48

### CODE REVIEW

**Comment by melvinheks** [comment]
Most of the line diffs are a result of fixing https://github.com/meowl-surveillance-system/meowl/issues/47 and https://github.com/meowl-surveillance-system/meowl/issues/48

File: src/web/client/src/App.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import Register from './components/Register/Register';
+import Login from './components/Login/Login';
+import Navbar from './components/Navbar/Navbar';
+
+interface Props {}
+interface State {
+  isLoggedIn: boolean
+}
+
+class App extends Component<Props, State> {
```
Comment (chc5) [inline src/web/client/src/App.tsx]: Comment class on top

File: src/web/client/src/components/Login/Login.test.tsx [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import React from 'react';
+import { render, screen, fireEvent } from '@testing-library/react';
+import Login from './Login';
+
+describe('Login component', () => {
+    test('renders Login component', () => {
```
Comment (chc5) [inline src/web/client/src/components/Login/Login.test.tsx]: There's no test here

File: src/web/client/src/components/Login/Login.tsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+interface Props {
+  isLoggedIn: boolean
+}
+
+interface State {
+  username: string;
+  password: string;
+}
+
+export default class Login extends Component<Props, State> {
```
Comment (chc5) [inline src/web/client/src/components/Login/Login.tsx]: Comment on top

File: src/web/client/src/components/Login/Login.tsx [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      username: "",
+      password: "",
+    };
+  }
+
+  handleChange = (event: ChangeEvent<HTMLInputElement>): void => {
```
Comment (chc5) [inline src/web/client/src/components/Login/Login.tsx]: I believe you should just handle this change with a one liner

File: src/web/client/src/components/Login/Login.tsx [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+    }
+    return (
+      <Container>
+        <Typography variant="h5" component="h6">Login</Typography>
+        <FormControl data-testid="form">
+          <TextField
+            id="username-input"
+            name="username"
+            value={this.state.username}
+            onChange={this.handleChange}
```
Comment (chc5) [inline src/web/client/src/components/Login/Login.tsx]: I suggest ```onChange={(username) => this.setState({username})}```

File: src/web/client/src/components/Login/Login.tsx [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+            placeholder="Username"
+            inputProps={{ "data-testid": "username-test" }}
+          />
+          <TextField
+            id="password-input"
+            name="password"
+            value={this.state.password}
+            onChange={this.handleChange}
```
Comment (chc5) [inline src/web/client/src/components/Login/Login.tsx]: I suggest onChange={(password) => this.setState({password})}

File: src/web/client/src/components/Navbar/Navbar.tsx [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+
+interface Props { isLoggedIn: boolean }
+interface State {}
+
+export default class Navbar extends Component<Props, State> {
+	logoutSubmit = (): void => {
+    	const requestOptions = {
+      		method: "POST",
+    	};
+    	fetch(`/auth/logout`, requestOptions)
```
Comment (chc5) [inline src/web/client/src/components/Navbar/Navbar.tsx]: Are there any response that you should be aware of if things go wrong

File: src/web/client/src/components/Navbar/Navbar.tsx [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+					<Toolbar>
+						<Typography variant="title">
+						Meowl
+						</Typography>
+						<Grid container alignItems="flex-start" justify="flex-end" direction="row">
+							<Link href="/" color="inherit" onClick={this.logoutSubmit}>Logout</Link>
+						</Grid>
+					</Toolbar>
+				</AppBar>
+			:
```
Comment (chc5) [inline src/web/client/src/components/Navbar/Navbar.tsx]: Please wrap these elements with parentheses

File: src/web/client/src/components/Register/Register.tsx [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+	constructor(props: Props) {
+		super(props);
+		this.state = {
+			email: "",
+			username: "",
+			password: ""
+		}
+	}
+
+	onChange = (event: ChangeEvent<HTMLInputElement>): void => {
```
Comment (chc5) [inline src/web/client/src/components/Register/Register.tsx]: I'm not sure but the method name is just unclear.

File: src/web/server/app/controllers/auth.ts [hunk 10]
```diff
@@ -3,6 +3,15 @@ import { v4 as uuidv4 } from 'uuid';
 
 import * as authServices from '../services/auth';
 
+export const isLoggedIn = (req: Request, res: Response) => {
```
Comment (chc5) [inline src/web/server/app/controllers/auth.ts]: any comments?

File: src/web/server/app/routes/auth.ts [hunk 11]
```diff
@@ -11,6 +11,10 @@ import * as authController from '../controllers/auth';
 
 const app = express();
 
+app.get('/isLoggedIn', (req: express.Request, res: express.Response) => {
```
Comment (chc5) [inline src/web/server/app/routes/auth.ts]: Comment

**Comment by yliu0715** [comment]
Closing this. Look at #74

**Pull request closed without merging**